content/en/platform/corda/4.8/open-source/generating-a-node-cordform.md

```diff
@@ -0,0 +1,303 @@
+
+A `Notary` node, which:
+
+* Provides a validating Notary service.
+* Runs the `corda-finance` CorDapp.
+
+`PartyA` and `PartyB` nodes, each of which:
+
+* Does not provide any services.
+* Runs the `corda-finance` CorDapp.
+* Has an RPC (Remote Procedure Call) user (`user1`), which enables you to log in the node via RPC.
+
+All three nodes also include any CorDapps defined in the project's source directories, even if these CorDapps are not listed in each node's `cordapps` setting. As a result, if you run the `deployNodes` task from the template CorDapp, for example, it will automatically build and add the template CorDapp to each node.
+
+{{< note >}}
+The three nodes described here are just an example. `Cordform` allows you specify any number of nodes and you can define their configurations and names as needed.
+{{< /note >}}
+
+The following example, as defined in the [Kotlin CorDapp Template](https://github.com/corda/cordapp-template-kotlin/blob/release-V4/build.gradle#L120), shows a `Cordform` task called `deployNodes` that creates the three nodes described above: `Notary`, `PartyA`, and `PartyB`.
+
+```groovy
+task deployNodes(type: net.corda.plugins.Cordform, dependsOn: ['jar']) {
+    directory "./build/nodes"
+    node {
+        name "O=Notary,L=London,C=GB"
+        // The Notary will offer a validating Notary service.
+        notary = [validating : true]
+        p2pPort  10002
+        rpcSettings {
+            port 10003
+            adminPort 10023
+        }
+        h2Port   10004
+        // Starts an internal SSH server providing a management shell on the node.
+        sshdPort 2223
+        // Includes the corda-finance CorDapp on our node.
+        cordapps = ["$corda_release_distribution:corda-finance:$corda_release_version"]
+        extraConfig = [
+            // Setting the JMX reporter type.
+            jmxReporterType: 'JOLOKIA',
+            // Setting the H2 address.
+            h2Settings: [ address: 'localhost:10030' ]
+        ]
+    }
+    node {
+        name "O=PartyA,L=London,C=GB"
+        p2pPort  10005
+        rpcSettings {
+            port 10006
+            adminPort 10026
+        }
+        h2Port   10008
+        cordapps = ["$corda_release_distribution:corda-finance:$corda_release_version"]
+        // Grants user1 all RPC permissions.
+        rpcUsers = [[ user: "user1", "password": "test", "permissions": ["ALL"]]]
+    }
+    node {
+        name "O=PartyB,L=New York,C=US"
+        p2pPort  10009
+        rpcSettings {
+            port 10010
+            adminPort 10030
+        }
+        h2Port   10012
+        cordapps = ["$corda_release_distribution:corda-finance:$corda_release_version"]
+        // Grants user1 the ability to start the MyFlow flow.
+        rpcUsers = [[ user: "user1", "password": "test", "permissions": ["StartFlow.net.corda.flows.MyFlow"]]]
+    }
+}
+```
+
+{{< note >}}
+Make sure to use Corda gradle plug-in version 5.0.10 or above.
```

Comment:
This is a requirement, not really a note. Fine if this is here as a reminder, but it should be laid out in the requirements, and made clear if it's important enough for a note later.